### PR TITLE
Change default log location

### PR DESCRIPTION
**Pull request by dszmigielski**
Change of default log location for opentelemetry distribution, so it doesn't overlap with datadog.

### CODE REVIEW

**Review by pellared — APPROVED** [review]
Approving.
It would be good if someone tests the Windows Installer manually.
There are still some stuff which should be renamed (e.g. in `.github/ISSUE_TEMPLATE/bug_report.md` and some comments in `src/Datadog.Trace.ClrProfiler.Native/environment_variables.h`) but we can have a separate PR for it.

File: src/Datadog.Trace.ClrProfiler.Native/pal.h [hunk 1]
```diff
@@ -58,7 +58,7 @@ inline WSTRING DatadogLogFilePath() {
   return ToWSTRING(program_data +
                    R"(\Datadog .NET Tracer\logs\dotnet-tracer-native.log)");
```
Comment (pellared) [inline src/Datadog.Trace.ClrProfiler.Native/pal.h:59]: I thonk this should be renamed as well

Comment (dszmigielski) [reply src/Datadog.Trace.ClrProfiler.Native/pal.h]: Fixed

**Review by pjanotti — APPROVED** [review]

**Comment by nrcventura** [comment]
I think that you'll also need to update some entries in Config.wxi, because I think that it's using some of those variables to define where things are installed.

**Review by zacharycmontoya — APPROVED** [review]

**Comment by zacharycmontoya** [comment]
> I think that you'll also need to update some entries in Config.wxi, because I think that it's using some of those variables to define where things are installed.

If this PR is just concerned with the logs directory, then the MSI doesn't need to be changed. The MSI only affects where the automatic instrumentation files are placed on disk (Program Files) which is separate from the logs which will exist in Program Data.

**Comment by pjanotti** [comment]
It still looks like we will have to revisit the wxs files but I want to merge to at least get Linux logs moved.

**Pull request merged**